File: Java/src/onreadyfyMe/AutoElectrico.java
package onreadyfyMe;

import java.text.DecimalFormat;

public class AutoElectrico extends Auto {
	private String voltaje;

	public String getVoltaje() {
		return voltaje;
	}

	public void setVoltaje(String voltaje) {
		this.voltaje = voltaje;
	}

	public AutoElectrico(String marca, String modelo, double precio, String puertas, String voltaje) {
		super(marca, modelo, precio, puertas);
		this.voltaje = voltaje;
	}

	public AutoElectrico() {
	}

	@Override
	public void mostrarTodosDatos() {
		DecimalFormat formatea = new DecimalFormat("###,###.00");
		System.out.println("Marca: " + getMarca() + " // Modelo: " + getModelo() + " // Puertas: " + getPuertas()
				+ " // Precio: $" + formatea.format(getPrecio()) + " // Voltaje: " + getVoltaje());
	}

	@Override
	public boolean esLujo() {
		boolean esLujo = false;
		if ((int) Double.parseDouble(getVoltaje().substring(0, 3)) >= 240) {
			return esLujo = true;
		}
		return esLujo;
	}
}

File: Python/auto_electrico.py
from auto import Auto

class AutoElectrico(Auto):
    __voltaje = ""

    def __init__(self, marca, modelo, precio, puertas, voltaje):
        Auto.__init__(self, marca, modelo, precio, puertas)
        self.__voltaje = voltaje

    def get_voltaje(self):
        return self.__voltaje

    def set_voltaje(self, voltaje):
        self.__voltaje = voltaje

    def __str__(self):
        return f"Marca: {self.get_marca()} // Modelo: {self.get_modelo()} // Puertas: {self.get_puertas()} // Precio: ${'{:,.2f}'.format(self.get_precio()).replace(',','~').replace('.',',').replace('~','.')} // Voltaje: {self.get_voltaje()}"

    def __gt__(self, auto):
        return (self.get_precio() > auto.get_precio())
    
    def esLujo(self):
        if int(self.get_voltaje()[:3]) >= 240:
            return True
        else:
            return False